### PR TITLE
[13.0][ADD] purchase_requisition_order_remaining_qty: This module allows you to define agreement types to use the remaining quantity of the agreement compared with the already proposed RfQs quantity.

### PR DESCRIPTION
**Pull request by victoralmau**
This module allows you to define the remaining quantity in the order lines created from a purchase order agreement.

Please @davidvidal and @pedrobaeza can you review it?

@Tecnativa TT33877

### CODE REVIEW

File: purchase_requisition_order_remaining_qty/README.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/purchase-workflow-13-0/purchase-workflow-13-0-purchase_requisition_order_remaining_qty
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/142/13.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module allows you to define the remaining quantity in the order lines created from a purchase order agreement.
```
Comment (pedrobaeza) [inline purchase_requisition_order_remaining_qty/README.rst]:
```suggestion
This module allows you to define agreement types to use the remaining quantity of the agreement compared with the already proposed RfQs quantity.
```

File: purchase_requisition_order_remaining_qty/README.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to **Purchase > Configuration > Purchase Agreement Types**
+#. Create or update some record and set "Quantities" to "Remaining Qty".
```
Comment (pedrobaeza) [inline purchase_requisition_order_remaining_qty/README.rst]: These 2 lines should go in configuration section, and see my later comment about the text of the option.

File: purchase_requisition_order_remaining_qty/models/purchase_requisition_line.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseRequisitionLine(models.Model):
+    _inherit = "purchase.requisition.line"
+
+    propose_qty = fields.Float(
```
Comment (pedrobaeza) [inline purchase_requisition_order_remaining_qty/models/purchase_requisition_line.py]: Remember to use the proper verb form:

```suggestion
    proposed_qty = fields.Float(
```

File: purchase_requisition_order_remaining_qty/models/purchase_requisition_line.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        "requisition_id.purchase_ids.order_line",
+        "requisition_id.purchase_ids.order_line.product_id",
+        "requisition_id.purchase_ids.order_line.product_qty",
+        "requisition_id.purchase_ids.state",
+    )
+    def _compute_propose_qty(self):
+        for item in self:
+            item.propose_qty = sum(
+                item.requisition_id.filtered(
+                    lambda x: x.type_id.quantity_copy == "remaining_qty"
```
Comment (pedrobaeza) [inline purchase_requisition_order_remaining_qty/models/purchase_requisition_line.py]: Why restrict to this type? The proposed quantity is going to be the same on any kind of selection, so let it computed for all.

File: purchase_requisition_order_remaining_qty/models/purchase_requisition_type.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class PurchaseRequisitionType(models.Model):
+    _inherit = "purchase.requisition.type"
+
+    quantity_copy = fields.Selection(
+        selection_add=([("remaining_qty", "Remaining Qty")])
```
Comment (pedrobaeza) [inline purchase_requisition_order_remaining_qty/models/purchase_requisition_type.py]: Put it on second place and using a similar redaction and avoid abbreviations on this kind of text that are not part of a column:

```suggestion
        selection_add=([("remaining_qty", "Use remaining quantities of agreement"), ("none", )])
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: purchase_requisition_order_remaining_qty/readme/CONFIGURE.rst [hunk 6]
```diff
@@ -0,0 +1,4 @@
+To configure this module, you need to:
+
+#. Go to **Purchase > Configuration > Purchase Agreement Types**
+#. Create or update some record and set "Quantities" to "Remaining Qty".
```
Comment (pedrobaeza) [inline purchase_requisition_order_remaining_qty/readme/CONFIGURE.rst]: Please check cross-changes:

```suggestion
#. Create or update some record and set "Quantities" to "Use remaining quantities of agreement".
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
One thing more: can we avoid the creation of lines with 0 qty at other level?

**Comment by victoralmau** [comment]
> One thing more: can we avoid the creation of lines with 0 qty at other level?

Done. Now order lines with quantity 0 are not created.

**Review by pedrobaeza — APPROVED** [review]

File: purchase_requisition_order_remaining_qty/models/purchase_requisition_type.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo import fields, models
+
+
+class PurchaseRequisitionType(models.Model):
+    _inherit = "purchase.requisition.type"
+
+    quantity_copy = fields.Selection(
+        selection_add=(
+            [("remaining_qty", "Use remaining quantities of agreement"), ("none",)]
```
Comment (chienandalu) [inline purchase_requisition_order_remaining_qty/models/purchase_requisition_type.py:12]: What's that `("none", )` for?

**Review by chienandalu — COMMENTED** [review]
Tested. Just a maybe silly doubt:

File: purchase_requisition_order_remaining_qty/models/purchase_requisition_type.py (hunk 7 above)
Comment (pedrobaeza) [reply purchase_requisition_order_remaining_qty/models/purchase_requisition_type.py:12]: It's the way to insert an element before another (and the selection has that confusing value `none`):
```python
selection = [('a', 'A'), ('b', 'B')] 
selection_add = [('c', 'C'), ('b',)] 
```
> result = [('a', 'A'), ('c', 'C'), ('b', 'B')]

Comment (chienandalu) [reply purchase_requisition_order_remaining_qty/models/purchase_requisition_type.py:12]: Ok! It rings my bell now :)

**Review by chienandalu — APPROVED** [review]

**Pull request merged**